File: 907-koko-eating-bananas/koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int left = 1;
        int right = *max_element(piles.begin(), piles.end());

        while(left <= right){
            int mid = (left+right)/2;
            long hours=0;
            for(int i=0; i<piles.size(); i++){
                hours += ceil((double)piles[i]/(double)mid);
            }
            if(hours <= h) right = mid-1;
            else left = mid+1;
        }
        return left;
    }
};

File: 907-koko-eating-bananas/koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles) # range for k

        while(l<=r):
            k = (l + r) // 2
            hours = 0
            for pile in piles:
                hours += ceil(pile/k)

            if hours <= h: r = k-1 
            else: l = k + 1

        return l